src/contributing/debug.md

```diff
@@ -0,0 +1,66 @@
+# Troubleshooting badly behaving apps
+
+Despite the best efforts of Void Linux maintainers, it is possible that you will
+have issues with packages from the official repositories. On such cases, there
+are some steps to follow in order to provide as complete a bug report as
+possible.
+
+Some of these steps can even help you find the cause of the issue yourself!
+
+## Look at error messages attentively THIS NEEDS WORK
+
+Python programs, for example, complain loudly about missing modules and other
+issues like that. If you're using a compiled binary, and it's missing libraries
+for some reason, your dynamic linker should tell you as well.
+
+## Check for issues in the package database
+
+You can use the `-a` flag for
+[xbps-pkgdb(1)](https://man.voidlinux.org/xbps-pkgdb.1) to run a complete check
+on all systems packages, which can detect any files that may have been altered
+when they shouldn't have been. For any of the files listed in this step, you
+should attempt to reinstall them with the `-f` flag for
+[xbps-install(1)](https://man.voidlinux.org/xbps-install.1). An example can be
+seen below:
+
+```
+# xbps-pkgdb -a
+ERROR: p7zip: hash mismatch for /usr/bin/7z.
+ERROR: p7zip: files check FAILED.
+# xbps-install -f p7zip
+```
+
+After this is done, check if the issue persists. It should be noted that issues
+like this can indicate hardware issues in your storage media or RAM.
```

Comment:
It might, but it also might not. I feel like this would lead to a lot of people going OH NO MY RAM IS DEAD when they actually powered off the computer in the middle of an upgrade or smth.

Reply:
Good point.